Drivers/Periphs/Src/SDCard.c

```diff
@@ -0,0 +1,104 @@
+
+#include <stdio.h>
+#include <string.h>
+#include <stdarg.h> //for va_list var arg functions
+#include "fatfs.h"
+#include "SDCard.h"
+#include "main.h"
+
+FRESULT fres;   //Result after operations
+
+UART_HandleTypeDef *huart;  //pointer to UART handler
+
+void SDCard_Init(UART_HandleTypeDef *uartBus){
+    huart = uartBus;
+}
+
+void myprintf(const char *fmt, ...){
+    static char buffer[256];
+    va_list args;
+    va_start(args, fmt);
+    vsnprintf(buffer, sizeof(buffer), fmt, args);
+    va_end(args);
+
+    int len = strlen(buffer);
+    HAL_UART_Transmit(huart, (uint8_t*)buffer, len, -1);
+}
+
+void SDCard_OpenFileSystem(FATFS FatFs){
+    //mount the drive
+    fres = f_mount(&FatFs, "", 1); //1=mount now
+    if (fres != FR_OK) {
+  	myprintf("f_mount error (%i)\r\n", fres);
+  	while(1);
```

Comment:
Is the desired behavior to just spin forever if we fail to mount the filesystem? Can we maybe return an error instead, if it is possible to recover from this?

Reply:
I have some of the functions in SDCard.c to return `ErrorStatus` in my latest commit